nibabel/testing/__init__.py

```diff
@@ -164,3 +166,11 @@ def __init__(self, *args, **kwargs):
         warnings.warn('catch_warn_reset is deprecated and will be removed in '
                       'nibabel v3.0; use nibabel.testing.clear_and_catch_warnings.',
                       FutureWarning)
+
+
+def runif_extra_has(test_str):
+    """Decorator checks to see if NIPY_EXTRA_TESTS env var contains test_str"""
+    def decorator(func):
+        skip_set = os.environ.get('NIPY_EXTRA_TESTS', '').split(',')
+        return skipif(test_str not in skip_set, "Skip slow tests.")(func)
+    return decorator
```

Comment:
This calls all extra tests "slow tests", and re-parses `NIPY_EXTRA_TESTS` each time. You could remove a nested function and gain a little brevity doing something like this:

At top:

    EXTRA_TESTS = set(os.environ.get('NIPY_EXTRA_TESTS', '').split(','))

New decorator:

    def runif_extra_has(test_str):
        """Decorator checks to see if NIPY_EXTRA_TESTS env var contains test_str"""
        return skipif(test_str not in EXTRA_TESTS, "Skip {0} tests.".format(test_str))

At the very least, I'd fix the `skipif` string.